pkg/reconciler/source/duck/controller.go

```diff
@@ -69,16 +66,16 @@ func NewController(crd string, gvr schema.GroupVersionResource, gvk schema.Group
 		}
 
 		r := &Reconciler{
-			Base:            reconciler.NewBase(ctx, controllerAgentName, cmw),
-			eventTypeLister: eventTypeInformer.Lister(),
-			crdLister:       crdInformer.Lister(),
-			sourceLister:    sourceLister,
-			gvr:             gvr,
-			crdName:         crd,
+			eventTypeLister:   eventTypeInformer.Lister(),
+			crdLister:         crdInformer.Lister(),
+			sourceLister:      sourceLister,
+			gvr:               gvr,
+			crdName:           crd,
+			eventingClientSet: eventingclient.Get(ctx),
 		}
-		impl := controller.NewImpl(r, r.Logger, ReconcilerName)
+		impl := controller.NewImpl(r, logging.FromContext(ctx), ReconcilerName)
```

Comment:
can you save logger := logging.FromContext(ctx) and use that here and below?

Reply:
in this case it is the same to pull it from context, we do this pattern in other places.

Reply:
but it is the same... so I did it just for you.